src/main/java/bio/terra/dao/DatasetDao.java

```diff
@@ -202,6 +208,13 @@ private Dataset retrieveWorker(String sql, MapSqlParameterSource params) {
         return summaries;
     }
 
+
+    public Integer total() {
```

Comment:
I'm concerned about this implementation. When we add in permission checks, how will we handle the enumeration and the total?

The total is going to touch every row to count it and we'll have to compute visibility for each row every time, right? For that, we can't use a simple count - we will need study data to compare to the user's group memberships.

With this implementation every enumerate is going to compute visibility in the enumerate AND here in the total. I think we need to find another way. If it was just Postgres, I wouldn't worry, because it caches well. But combining the permission check could make it slow.

For now, I think we should make the enumerate interface return both the list and the total instead of parts, so we have the opportunity to change the algorithm without changing the interface.

Reply:
Good point. I was worried about that too. When we filter based on access we need to filter the results where there is a match in `https://sam.dsde-dev.broadinstitute.org/api/resources/v1/dataset` (or /studies). 

I agree that combining it into one call for total and enumerate is cleaner and will speed things up by only requiring one SAM call. Thanks